profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/SnapshotApiMockTest.java

```diff
@@ -0,0 +1,162 @@
+      assertEquals(list.get(0).properties().name(), "test snapshot");
+      
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/snapshots?depth=5");
+   }
+   
+   
+   @Test
+   public void testGetListWith404() throws InterruptedException {
+      server.enqueue(response404());
+      List<Snapshot> list = snapshotApi().getList(new DepthOptions().depth(1));
+      assertTrue(list.isEmpty());
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/snapshots?depth=1");
+   }
+    
+   @Test
+   public void testGetSnapshot() throws InterruptedException {
+      MockResponse response = new MockResponse();
+      response.setBody(stringFromResource("/snapshot/get.json"));
+      response.setHeader("Content-Type", "application/vnd.profitbricks.resource+json");
+      
+      server.enqueue(response);
+      
+      Snapshot snapshot = snapshotApi().getSnapshot("some-id");
+      
+      assertNotNull(snapshot);
+      assertEquals(snapshot.properties().name(), "snapshot desc...");
+      
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/snapshots/some-id");
+   }
+   
+   @Test
+   public void testGetSnapshotWithDepth() throws InterruptedException {
+      MockResponse response = new MockResponse();
+      response.setBody(stringFromResource("/snapshot/get-depth-5.json"));
+      response.setHeader("Content-Type", "application/vnd.profitbricks.resource+json");
+      
+      server.enqueue(response);
+      
+      Snapshot snapshot = snapshotApi().getSnapshot("some-id", new DepthOptions().depth(5));
+      
+      assertNotNull(snapshot);
+      assertEquals(snapshot.properties().name(), "test snapshot 2");
+      
+      assertEquals(this.server.getRequestCount(), 1);
+      assertSent(this.server, "GET", "/snapshots/some-id?depth=5");
+   }
+   
+   
+   public void testGetSnapshotWith404() throws InterruptedException {
+      server.enqueue(response404());
+
+      Snapshot snapshot = snapshotApi().getSnapshot("some-id");
+      
+      assertEquals(snapshot, null);
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/snapshots/some-id");
+   }   
+  
+   @Test
+   public void testUpdate() throws InterruptedException {
+      server.enqueue(
+         new MockResponse().setBody(stringFromResource("/snapshot/get.json"))
+      );
+      
+      api.snapshotApi().updateSnapshot(
+              Snapshot.Request.updatingBuilder()
+              .id("some-id")
+              .name("new-snapshot-name")
+              .description("description...")
+              .build());
+            
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "PATCH", "/rest/snapshots/some-id", "{\"name\": \"new-snapshot-name\", \"description\": \"description...\"}");
+   }
+   
+   @Test
+   public void testDelete() throws InterruptedException {
+      server.enqueue(
+         new MockResponse().setBody("")
+      );
```

Comment:
Instead of an empty body, wouldn't it be better to configure a response with the 204 status code?